### PR TITLE
Add gh-action workflow, drop "docs" directory.

### PR DESCRIPTION
**Pull request by olifre**
This drops the `docs` subdirectory and adds a workflow which builds the static page instead, pushing the result to a separate `gh-pages` branch if any change is pushed to the `master` branch. 
Hence, after changing repo configuration to publish the `gh-pages` branch instead of the `docs` directory, the Hugo site should be build automatically via the action :smile:.

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+      - name: Setup Hugo
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: 'latest'
+          extended: true
+
+      - name: Build
+        run: |
+          hugo --minify
```
Comment (Vindaar) [inline .github/workflows/gh-pages.yml]: This needs to include the theme we use:
```suggestion
          hugo --minify --theme book
```

**Review by Vindaar — CHANGES_REQUESTED** [review]
Thanks a lot! :)

As mentioned as a comment on the YAML file, we need to tell hugo which theme to use.

I just typed a bunch of stuff here saying that I don't know how things like themes are handled, i.e. if it should be cloned or used as a submodule. Then I checked and saw that I'm already using a submodule for the theme, haha. Apparently that's how themes are handled in Hugo land typically (which I suppose makes sense).

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (olifre) [reply .github/workflows/gh-pages.yml]: Fixed, thanks! :smile: .

**Comment by olifre** [comment]
@Vindaar , I also started with `pandoc` on top of this PR here:
https://github.com/olifre/MHB_test/commit/2491625157bebb94a1affe4558b04d8c9d15fbc9
However, the `_index.md` files we currently have are of course not too useful here. Do you have a good idea on how to add `pandoc` best? Can `hugo` be told to "pre-process" the markdown files?

**Comment by Vindaar** [comment]
> @Vindaar , I also started with `pandoc` on top of this PR here:
> [olifre@2491625](https://github.com/olifre/MHB_test/commit/2491625157bebb94a1affe4558b04d8c9d15fbc9)
> However, the `_index.md` files we currently have are of course not too useful here. Do you have a good idea on how to add `pandoc` best? Can `hugo` be told to "pre-process" the markdown files?

I opened a new issue here: #2.

**Comment by Vindaar** [comment]
Let's see if this works. ;)

**Comment by olifre** [comment]
> Let's see if this works. ;)

Looks good, only the actual github pages deployment failed — probably there's still `/docs` in the settings? 
The way the action sets up the branch is that the content arrives directly to `gh-pages` in `/` ;-).

**Comment by Vindaar** [comment]
> > Let's see if this works. ;)
> 
> Looks good, only the actual github pages deployment failed — probably there's still `/docs` in the settings?
> The way the action sets up the branch is that the content arrives directly to `gh-pages` in `/` ;-).

Yes, I only changed the (master, docs) ⇒ (gh-pages, /root) after I merged. Unfortunately it doesn't seem like it realizes that that's outdated (and "rerunning" the "Page build" action doesn't do anything. I assume from the next commit it should work fine. 
The  website itself works fine (from gh-pages branch now).

**Comment by olifre** [comment]
> I assume from the next commit it should work fine.
> The website itself works fine (from gh-pages branch now).

Indeed, GitHub Pages seems to take long and finally show "abandoned" right now for some runs. Might be related to their ongoing issues with GitHub Packages (see https://www.githubstatus.com/ ), probably the underlying storage is similar.

**Pull request merged**